counters.hh

```diff
@@ -88,7 +88,14 @@ public:
 
 using counter_shard_view = basic_counter_shard_view<mutable_view::no>;
 
-std::ostream& operator<<(std::ostream& os, counter_shard_view csv);
+template <>
+struct fmt::formatter<counter_shard_view> : fmt::formatter<std::string_view> {
+    template <typename FormatContext>
+    auto format(const counter_shard_view& csv, FormatContext& ctx) const {
+        return fmt::format_to(ctx.out(), "{{global_shard id: {} value: {}, clock: {}}}",
+                              csv.id(), csv.value(), csv.logical_clock());
+    }
```

Comment:
It's really unfortunate that fmt forces the formatter to live in the header. I guess it's one price for compile-time format-string checking.

Reply:
@avikivity after rereading https://stackoverflow.com/questions/63075693/custom-type-fmt-formatters-with-external-termplates-any-drawbacks and {fmt}'s source code, i think we can indeed move at least the `format()` part into .cc file. i will apply this change to all existing formatters as long as they are non-trivial. by trivial, i mean just a delegation to yet another formatter.

Reply:
the major use case of compile-time format-string checking is Seastar's logging facility. but it is still pending on review. see https://github.com/scylladb/seastar/pull/1196. we are using runtime- format string at this moment. to get `parse()` implementations accessible at build-time, it does seems that we need to have them in the header file. but the `format()` implementations can be moved into .cc file.